record_layer.go

```diff
@@ -77,12 +77,12 @@ func unpackDatagram(buf []byte) ([][]byte, error) {
 
 	for offset := 0; len(buf) != offset; {
 		if len(buf)-offset <= recordLayerHeaderSize {
-			return nil, errDTLSPacketInvalidLength
+			return nil, errInvalidPacketLength
 		}
 
 		pktLen := (recordLayerHeaderSize + int(binary.BigEndian.Uint16(buf[offset+11:])))
 		if offset+pktLen > len(buf) {
-			return nil, errLengthMismatch
+			return nil, errInvalidPacketLength
```

Comment:
Why `errInvalidPackageLength` instead of `errLengthMismatch`?

Reply:
There were two usage of `errLengthMismatch`.
One is caused by internal implementation, which must be treated as a fatal error.
Another one is caused by incoming data and DTLS must ignore error due to incoming data as written in RFC. https://tools.ietf.org/html/rfc6347#section-4.1.2.7